src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -220,6 +224,13 @@ protected void implCloseChannel() throws IOException {
 
     @Override
     public int read(ByteBuffer dst) throws IOException {
+        if (jfrTracing) {
+            return traceReadImpl(dst);
+        }
+        return readImpl(dst);
+    }
+
+    private int readImpl(ByteBuffer dst) throws IOException {
```

Comment:
Would you mind renaming these to implXXX rather than XXXImpl? The implXXX is the convention in this area (including in the API).

For placement, the read methods are grouped, and the write methods are grouped, only to avoid needing to jump around the file when touching this code. So I think easier if the methods that wraps implRead be located with the read methods.

Reply:
Will fix.